api/shared/compare/serializers.py

```diff
@@ -76,3 +90,172 @@ class FlagComparisonSerializer(serializers.Serializer):
+        base_name = impacted_file.base_name
+        head_name = impacted_file.head_name
+        return base_name is not None and head_name is None
+
+    def get_change_coverage(
+        self, impacted_file: ImpactedFile
+    ) -> serializers.FloatField:
+        return impacted_file.change_coverage
+
+    def get_misses_count(self, impacted_file: ImpactedFile) -> serializers.IntegerField:
+        return impacted_file.misses_count
+
+    def get_hashed_path(self, impacted_file: ImpactedFile) -> serializers.CharField:
+        path = impacted_file.head_name
+        encoded_path = path.encode()
+        md5_path = hashlib.md5(encoded_path)
+        return md5_path.hexdigest()
+
+    def get_segments(
+        self, impacted_file: ImpactedFile
+    ) -> ImpactedFileSegmentSerializer:
+        file_comparison = self.context["comparison"].get_file_comparison(
+            impacted_file.head_name, with_src=True, bypass_max_diff=True
+        )
+        return [
+            ImpactedFileSegmentSerializer(segment).data
+            for segment in file_comparison.segments
+        ]
+
+
+class ImpactedFilesComparisonSerializer(ComparisonSerializer):
+
+    files = serializers.SerializerMethodField()
+
+    def get_files(self, comparison: Comparison) -> List[dict]:
+        comparison_table = CommitComparison._meta.db_table
+        commit_table = Commit._meta.db_table
+        commit_comparison = CommitComparison.objects.raw(
+            f"""
+            select
+                {comparison_table}.*,
+                base_commit.commitid as base_commitid,
+                compare_commit.commitid as compare_commitid
+            from {comparison_table}
+            inner join {commit_table} base_commit
+                on base_commit.id = {comparison_table}.base_commit_id and base_commit.repoid = {comparison.base_commit.repository_id}
+            inner join {commit_table} compare_commit
+                on compare_commit.id = {comparison_table}.compare_commit_id and compare_commit.repoid = {comparison.head_commit.repository_id}
+            where (base_commit.commitid, compare_commit.commitid) in %s
+        """,
+            [
+                tuple(
+                    [(comparison.base_commit.commitid, comparison.head_commit.commitid)]
+                )
+            ],
+        ).using("default")
+
+        # Can't use pre-computed impacted files from CommitComparison
+        # first trigger a Celery task to create a comparison for this commit pair for the future
+        # then will fall back to retrieving and generating all files on the fly
+        if not commit_comparison:
+            try:
+                new_comparison = CommitComparison(
+                    base_commit=comparison.base_commit.commitid,
+                    compare_commit=comparison.head_commit.commitid,
+                )
+                new_comparison.save()
+                TaskService().compute_comparison(new_comparison.pk)
+            except Exception as e:
```

Comment:
Avoid catchall exceptions like this if possible since it can easily hide bugs.  Is there a specific error that you expect to happen in this case?